src/main/kotlin/com/zepben/evolve/cim/iec61970/base/core/NameType.kt

```diff
@@ -23,24 +23,45 @@ import com.zepben.evolve.services.common.extensions.asUnmodifiable
 class NameType(val name: String) {
 
     private var namesIndex: MutableMap<String, Name> = mutableMapOf()
-    private var namesMultiIndex: MutableMap<String, MutableList<Name>> = mutableMapOf()
+    private var namesMultiIndex: MutableMap<String, MutableMap<IdentifiedObject, Name>> = mutableMapOf()
 
     var description: String = ""
 
     /**
      * All names of this type.
      */
-    val names: Sequence<Name> get() = namesMultiIndex.values.asSequence().flatten() + namesIndex.values.asSequence()
+    val names: Sequence<Name> get() = namesMultiIndex.flatMap { it.value.values }.asSequence() + namesIndex.values.asSequence()
```

Comment:
todo: The change you have made to this function has a hidden collect to `List` due to the order of operations you have applied.

Can you please update it to `namesMultiIndex.asSequence().flatMap { it.value.values }`

Reply:
Done